jerry-core/ecma/builtin-objects/ecma-builtin-function-prototype.cpp

```diff
@@ -226,7 +226,151 @@ ecma_builtin_function_prototype_object_bind (ecma_value_t this_arg, /**< this ar
+    bound_this_prop_p = ecma_create_internal_property (function_p, ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_BOUND_THIS);
+
+    size_t arg_list_size;
+
+    if (arguments_number > 0)
+    {
+      bound_this_prop_p->u.internal_property.value = ecma_copy_value (arguments_list_p[0], false);
+
+      arg_list_size = (size_t) (arguments_number) * sizeof (ecma_value_t);
+    }
+    else
+    {
+      ecma_object_t *glob_obj_p = ecma_builtin_get (ECMA_BUILTIN_ID_GLOBAL);
+      bound_this_prop_p->u.internal_property.value = ecma_copy_value (ecma_make_object_value (glob_obj_p), false);
+      ecma_deref_object (glob_obj_p);
+
+      arg_list_size = (size_t) sizeof (ecma_value_t);
+    }
+
+    /* 9. */
+    ecma_value_t *arg_list = static_cast <ecma_value_t *> (mem_heap_alloc_block (arg_list_size,
+                                                                                 MEM_HEAP_ALLOC_SHORT_TERM));
+
+    ecma_number_t *count = ecma_alloc_number ();
+    *count = ecma_uint32_to_number (arguments_number);
+
+    if (arguments_number > 0)
+    {
+      (*count)--;
+    }
+
+    /*
+     * Set the first element of the arguments array to the count of optional arguments.
+     * We need the number of bounded arguments, when the target function is called.
+     */
+    arg_list[0] = ecma_make_number_value (count);
+
+    for (uint32_t i = 1; i < arguments_number; i++)
+    {
+      arg_list[i] = ecma_copy_value (arguments_list_p[i], false);
+    }
+
+    ecma_property_t *bound_args_prop_p;
+    bound_args_prop_p = ecma_create_internal_property (function_p, ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_BOUND_ARGS);
+    ECMA_SET_POINTER (bound_args_prop_p->u.internal_property.value, arg_list);
+
+    /*
+     * [[Class]] property is not stored explicitly for objects of ECMA_OBJECT_TYPE_FUNCTION type.
+     *
+     * See also: ecma_object_get_class_name
+     */
+
+    /* 17. */
+    ecma_number_t *length_p = ecma_alloc_number ();
+    *length_p = 0;
+
+    ecma_property_descriptor_t length_prop_desc = ecma_make_empty_property_descriptor ();
+    length_prop_desc.is_value_defined = true;
+    length_prop_desc.value = ecma_make_number_value (length_p);
+
+    length_prop_desc.is_writable_defined = true;
+    length_prop_desc.is_writable = false;
+
+    length_prop_desc.is_enumerable_defined = true;
+    length_prop_desc.is_enumerable = false;
+
+    length_prop_desc.is_configurable_defined = true;
+    length_prop_desc.is_configurable = false;
+
+    ecma_string_t *magic_string_length_p = ecma_get_magic_string (LIT_MAGIC_STRING_LENGTH);
+    ecma_completion_value_t completion = ecma_op_object_define_own_property (function_p,
+                                                                             magic_string_length_p,
+                                                                             &length_prop_desc,
+                                                                             false);
+
+    ecma_free_completion_value (completion);
```

Comment:
As `is_throw` argument is false, the value should always be simple completion value, and so, should not freed.
Please, replace the line with assertion like `ecma_is_completion_value_normal_true (completion)` or `ecma_is_completion_value_normal_true (completion) || ecma_is_completion_value_normal_false (completion)`.